uosc.lua

```diff
@@ -1674,15 +1676,28 @@ function render_timeline(this)
 	end
 
 	if (this.proximity_raw == 0 or this.pressed) and not (elements.speed and elements.speed.dragging) then
-		-- Hovered time
+		-- Hovered time and chapter
 		local hovered_seconds = state.duration * (cursor.x / display.width)
-		local box_half_width_guesstimate = (this.font_size * 4.2) / 2
+		local chapter_title = ''
+		if (state.chapters and options.chapters ~= 'none') then
+			for i = #state.chapters, 1, -1 do
+				local chapter = state.chapters[i]
+				if hovered_seconds >= chapter.time then
+					chapter_title = chapter.title
+					break
+				end
+			end
+		end
+		local time_formatted = mp.format_time(hovered_seconds)
+		local text_len = math.max(time_formatted:len(), chapter_title:len())
+		local box_half_width_estimate = text_width_estimate(text_len, this.font_size) / 2
 		ass:new_event()
 		ass:append('{\\blur0\\bord1\\shad0\\1c&H'..options.color_background_text..'\\3c&H'..options.color_background..'\\fn'..config.font..'\\fs'..this.font_size..bold_tag..'')
 		ass:append(ass_opacity(math.min(options.timeline_opacity + 0.1, 1)))
-		ass:pos(math.min(math.max(cursor.x, box_half_width_guesstimate), display.width - box_half_width_guesstimate), fay)
+		ass:pos(math.min(math.max(cursor.x, box_half_width_estimate), display.width - box_half_width_estimate), fay)
 		ass:an(2)
-		ass:append(mp.format_time(hovered_seconds))
+		ass:append(chapter_title .. '\\N')
```

Comment:
I wonder if it would be better to show them on separate ass lines so that they could be left- and right-margined independently, i.e. the timestamp would be closer to the hovered area for long titles.

Illustrations, where `o` represents the currently hovered position, i.e. `cursor.x`:

*current code*
```
this is a very long chapter title
            00:00:00
|o---------------|-----------------|------|

          this is a very long chapter title
                      00:00:00
|----------------|-----------------|-----o|
```

*my suggestion*
```
this is a very long chapter title
00:00:00
|o---------------|-----------------|------|

     this is a very long chapter title
                 00:00:00
|o---------------|---o-------------|------|

          this is a very long chapter title
                                   00:00:00
|o---------------|-----------------|-----o|
```

Reply:
Good idea.